tests/decostand-tests.R

```diff
@@ -1,50 +1,89 @@
-        
+###<--- BEGIN TESTS --->
+suppressPackageStartupMessages(require(vegan))
+
 ############################### CLR ####################################
 # Calculates clr-transformation. Should be equal.
 
 # Test data
-# data(varespec)
-testdata <- matrix(round(runif(1000, 0, 100)), nrow=20)
+data(varespec)
+testdata <- matrix(round(runif(1000, 0, 100)), nrow = 20)
 testdata <- testdata - 50
 testdata[testdata < 0] <- 0
 rownames(testdata) <- paste0("row", seq_len(nrow(testdata)))
 colnames(testdata) <- paste0("col", seq_len(ncol(testdata)))
 testdata.with.pseudo <- testdata + 1
 
 # Calculates relative abundance table
-relative <- vegan::decostand(testdata, "total")
-relative.with.pseudo <- vegan::decostand(testdata+1, "total")
+relative <- decostand(testdata, "total")
+relative.with.pseudo <- decostand(testdata + 1, "total")
 
 # CLR data
-x.clr <- vegan::decostand(testdata+1, method = "clr")
-x.rclr <- vegan::decostand(testdata, method = "rclr")
-x.clr.pseudo <- vegan::decostand(testdata, method = "clr", pseudocount=1)
+x.clr  <- decostand(testdata + 1, method = "clr")
+x.rclr <- decostand(testdata, method = "rclr")
+x.clr.pseudo <- decostand(testdata, method = "clr", pseudocount = 1)
 
-max(abs(x.clr - x.clr.pseudo))<1e-6
-max(abs(vegan::decostand(testdata+1, method = "clr", pseudocount=0)-x.clr.pseudo))<1e-6
+max(abs(x.clr - x.clr.pseudo)) < 1e-6
+max(abs(decostand(testdata + 1, method = "clr", pseudocount = 0) - x.clr.pseudo)) < 1e-6
 
 # Tests clr
-alt.clr <- t(apply(as.matrix(relative.with.pseudo), 1, FUN=function(x){
-log(x) - mean(log(x))}))
-max(abs(x.clr-alt.clr)) < 1e-6      
-all((x.rclr==0) == (testdata==0))
+alt.clr <- t(apply(as.matrix(relative.with.pseudo), 1, FUN = function(x){log(x) - mean(log(x))}))
+max(abs(x.clr - alt.clr)) < 1e-6      
+
+# Test that NAs are handled as expected in CLR
+x <- testdata
+x[sample(prod(dim(x)), 50)] <- NA # insert some NAs in the data
+# NAs in the original data remain NAs in the returned data
+all(is.na(decostand(x, "clr", na.rm = FALSE, pseudocount = 1)[is.na(x)])) == TRUE # NAs
+# For the other (non-NA) values, we get non-NA values back
+any(is.na(decostand(x, "clr", na.rm = FALSE, pseudocount = 1)[!is.na(x)])) == FALSE
+any(is.na(decostand(x, "clr", MARGIN = 2, na.rm = FALSE, pseudocount = 1)[!is.na(t(x))])) == FALSE 
```

Comment:
This is not returning `TRUE`:

```{r}
> any(is.na(decostand(x, "clr", MARGIN = 2, na.rm = FALSE, pseudocount = 1)[!is.na(t(x))])) == FALSE
[1] FALSE
```

Is this expected?

Reply:
Despite all the efforts to check each line, this had somehow escaped my attention.. good catch. This should be

`any(is.na(decostand(x, "clr", MARGIN = 2, na.rm = FALSE, pseudocount = 1)[!is.na(x)])) == FALSE` 

i.e. no transpose around `is.na(x)`.

Reply:
You can change, or I update the PR?

Reply:
I'll change; I have a few more lines to check with you that are causing some output diffs for me but I won't get to those until later this evening or first thing tomorrow.